homeassistant/components/zwave/__init__.py

```diff
@@ -399,11 +411,20 @@ def value_added(node, value):
                 value.enable_poll(polling_intensity)
             else:
                 value.disable_poll()
+            platform = get_platform(component, DOMAIN)
+            device = platform.get_device(
+                node=node, value=value, node_config=node_config, hass=hass)
+            if device:
+                dict_id = value.value_id
+
+                @callback
+                def discover_device(component, device, dict_id):
+                    """Put device in a dictionary and call discovery on it."""
+                    hass.data[DATA_ZWAVE_DICT][dict_id] = device
+                    discovery.load_platform(hass, component, DOMAIN, {
```

Comment:
We should probably use the async equivalent here, since we're already in the event loop at this point. No sense in creating a new job: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/helpers/discovery.py#L136